integration_tests.py

```diff
@@ -0,0 +1,234 @@
+
+    def test_a_iterative(self):
+        c = u"./zdns/zdns A --iterative"
+        name = u"zdns-testing.com"
+        cmd, res = self.run_zdns(c, name)
+        self.assertSuccess(res, cmd)
+        self.assertEqualAnswers(res, self.ROOT_A_ANSWERS, cmd)
+
+    def test_aaaa(self):
+        c = u"./zdns/zdns AAAA"
+        name = u"zdns-testing.com"
+        cmd, res = self.run_zdns(c, name)
+        self.assertSuccess(res, cmd)
+        self.assertEqualAnswers(res, self.ROOT_AAAA_ANSWERS, cmd)
+
+    def test_aaaa_iterative(self):
+        c = u"./zdns/zdns AAAA --iterative"
+        name = u"zdns-testing.com"
+        cmd, res = self.run_zdns(c, name)
+        self.assertSuccess(res, cmd)
+        self.assertEqualAnswers(res, self.ROOT_AAAA_ANSWERS, cmd)
+
+    def test_mx(self):
+        c = u"./zdns/zdns MX"
+        name = u"zdns-testing.com"
+        cmd, res = self.run_zdns(c, name)
+        self.assertSuccess(res, cmd)
+        self.assertEqualAnswers(res, self.MX_SERVERS, cmd)
+
+    def test_mx_iterative(self):
+        c = u"./zdns/zdns MX --iterative"
+        name = u"zdns-testing.com"
+        cmd, res = self.run_zdns(c, name)
+        self.assertSuccess(res, cmd)
+        self.assertEqualAnswers(res, self.MX_SERVERS, cmd)
+
+    def test_ns(self):
+        c = u"./zdns/zdns NS"
+        name = u"zdns-testing.com"
+        cmd, res = self.run_zdns(c, name)
+        self.assertSuccess(res, cmd)
+        self.assertEqualAnswers(res, self.NS_SERVERS, cmd)
+
+    def test_ns_iterative(self):
+        c = u"./zdns/zdns NS --iterative"
+        name = u"zdns-testing.com"
+        cmd, res = self.run_zdns(c, name)
+        self.assertSuccess(res, cmd)
+        self.assertEqualAnswers(res, self.NS_SERVERS, cmd)
+
+    def test_mx_lookup(self):
+        c = u"./zdns/zdns mxlookup --ipv4-lookup --ipv6-lookup"
+        name = u"zdns-testing.com"
+        cmd, res = self.run_zdns(c, name)
+        self.assertSuccess(res, cmd)
+        self.assertEqualMXLookup(res, self.MX_LOOKUP_ANSWER)
+
+    def test_mx_lookup_iterative(self):
+        c = u"./zdns/zdns mxlookup --ipv4-lookup --ipv6-lookup --iterative"
+        name = u"zdns-testing.com"
+        cmd, res = self.run_zdns(c, name)
+        self.assertSuccess(res, cmd)
+        self.assertEqualMXLookup(res, self.MX_LOOKUP_ANSWER)
+
+    def test_a_lookup(self):
+        c = u"./zdns/zdns alookup --ipv4-lookup --ipv6-lookup"
+        name = u"www.zdns-testing.com"
+        cmd, res = self.run_zdns(c, name)
+        self.assertSuccess(res, cmd)
+        self.assertEqualALookup(res, self.ALOOKUP_WWW_ZDNS_TESTING)
+
+    def test_a_lookup_iterative(self):
+        c = u"./zdns/zdns alookup --ipv4-lookup --ipv6-lookup --iterative"
+        name = u"www.zdns-testing.com"
+        cmd, res = self.run_zdns(c, name)
+        self.assertSuccess(res, cmd)
+        self.assertEqualALookup(res, self.ALOOKUP_WWW_ZDNS_TESTING)
+
+
+if __name__ == '__main__':
```

Comment:
While I agree that unit tests would be helpful in this code base, these provide coverage over core functionality, and there isn't any benefit of writing a test handler in go here instead of python. Go-based tests are not inherently better.